code/game/objects/items/storage/backpack.dm

```diff
@@ -889,6 +893,24 @@ GLOBAL_LIST_EMPTY_TYPED(radio_packs, /obj/item/storage/backpack/marine/satchel/r
 	fuel_type = "utnapthal"
 	has_gamemode_skin = TRUE
 
+//this is to revert change for the backpack that are for flametrower usage.
+// so that they can use custom mix to refill those backpack
+/obj/item/storage/backpack/marine/engineerpack/flamethrower/afterattack(obj/target, mob/user, proximity)
+	if(!proximity)
+		return
+	if (!(istype(target, /obj/structure/reagent_dispensers/fueltank)))
+		return
+
+	if (reagents.total_volume >= max_fuel)
+		to_chat(user, SPAN_NOTICE("The pack is already full!"))
+		return
+
+	if(reagents.total_volume < max_fuel)
+		target.reagents.trans_to(src, max_fuel)
+		to_chat(user, SPAN_NOTICE("You crack the cap off the top of the pack and fill it back up again from the tank."))
+		playsound(loc, 'sound/effects/refill.ogg', 25, TRUE, 3)
+		return
+
```

Comment:
Copied from `/obj/item/ammo_magazine/flamer_tank`:
```suggestion

/obj/item/storage/backpack/marine/engineerpack/flamethrower/verb/remove_reagents()
    set name = "Empty canister"
    set category = "Object"

    set src in usr

    if(usr.get_active_hand() != src)
        return

    if(alert(usr, "Do you really want to empty out [src]?", "Empty canister", "Yes", "No") != "Yes")
        return

    reagents.clear_reagents()

    to_chat(usr, SPAN_NOTICE("You empty out [src]"))
    update_icon()
```
Can be either `/obj/item/storage/backpack/marine/engineerpack/verb/remove_reagents()` or `/obj/item/storage/backpack/marine/engineerpack/flamethrower/verb/remove_reagents()` if you don't want it on the two welder packs.

Reply:
i used your code thanks